lib/linter/linter.js

```diff
@@ -919,7 +919,7 @@ function runRules(sourceCode, configuredRules, ruleMapper, parserOptions, parser
                         }
                         const problem = reportTranslator(...args);
 
-                        if (problem.fix && rule.meta && !rule.meta.fixable) {
+                        if (problem.fix && !(rule.meta && rule.meta.fixable)) {
                             throw new Error("Fixable rules should export a `meta.fixable` property.");
```

Comment:
Now that we throw an error in `Linter` rather than just `RuleTester`, "must" seems appropriate here. This wording will match #14573 as well. I'd also be fine extending the message to specify "code" and "whitespace" as allowable values.

```suggestion
                            throw new Error("Fixable rules must set the `meta.fixable` property.");
```

Reply:
Good suggestions! I'll fix this.

>  I'd also be fine extending the message to specify "code" and "whitespace" as allowable values.

Maybe we could also add the validation that the value is indeed `"code"` or `"whitespace"`? That wasn't discussed in https://github.com/eslint/eslint/issues/13349, and we're currently not using `"code"`/`"whitespace"` for anything, but since we're already making a breaking change this might be a good time to align the rules with the specification (and maybe we will use `"code"`/`"whitespace"` in the future to optimize autofixing).

Reply:
The earliest mention I could find of `fixable` is https://github.com/eslint/eslint/issues/5417#issuecomment-190475250. Since we haven’t started using it in five years, I’m not sure starting to enforce it strictly with a validation error now has a corresponding benefit to make the breaking change worth it.

Reply:
> The earliest mention I could find of `fixable` is [#5417 (comment)](https://github.com/eslint/eslint/issues/5417#issuecomment-190475250). Since we haven’t started using it in five years, I’m not sure starting to enforce it strictly with a validation error now has a corresponding benefit to make the breaking change worth it.

Agreed!

Changed the error message per suggestions in https://github.com/eslint/eslint/pull/14634/commits/7a983f9a1d309b22ac657ad0a6f9aba741d77516